src/db.py

```diff
@@ -741,6 +812,10 @@ def _upgrade(oldversion):
                 # player id as a string). When nocasing players, this may cause some records to be merged.
                 with open(os.path.join(dn, "db", "upgrade2.sql"), "rt") as f:
                     c.executescript(f.read())
+            if oldversion < 3:
+                print ("Upgrade from version 2 to 3...", file=sys.stderr)
+                with open(os.path.join(dn, "db", "upgrade3.sql"), "rt") as f:
+                    c.executescript(f.read())
```

Comment:
Can't you possibly put that in a loop? e.g.:

```
assert oldversion >= 1
for i in range(2, SCHEMA_VERSION+1):
    if oldversion < i:
        print("Upgrading from version {0} to {1}...".format(oldversion, i), file=sys.stderr)
        with open(os.path.join(dn, "db", "upgrade{0}.sql".format(i)), "rt") as f:
            c.executescript(f.read())
        oldversion = i
```

That way you don't have to keep adding identical blocks everytime you perform an update. As for the comments about what each update does, these can go above the loop, e.g. by saying "Upgrade 2 change x, y, and z" and so on.